File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        Dictionary<int, int> countMap = new Dictionary<int, int>();
        foreach(int i in tangerine)
        {
            if(countMap.ContainsKey(i))
            {
                countMap[i] = countMap[i] + 1;
            }
            else
            {
                countMap[i] = 1;
            }
        }
        
        List<int> counts = countMap
            .Values
            .OrderByDescending((int x) => x)
            .ToList();
        
        foreach(int cnt in counts)
        {
            answer = answer + 1;
            k = k - cnt;
            if (k <= 0)
            {
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;

    vector<int> box(10000001);

    for (int i : tangerine) {
        box[i]++;
    }

    sort(box.rbegin(), box.rend());

    for (int i = 0; i < box.size(); i++) {
        if (box[i] < k) {
            answer++;
            k -= box[i];
        }
        else {
            answer++;
            break;
        }
    }
    

    return answer;
}
